Assets/General/Scripts/Story/StoryManager.cs

```diff
@@ -0,0 +1,176 @@
+/*****************************************************************************
+// Name :           StoryManager.cs
+// Author :         Charlie Polonus
+// Created :        11/6/2024
+// Description :    Handles the story beats and story beat events for during
+                    the game
+*****************************************************************************/
+
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+public class StoryManager : MonoBehaviour
+{
+    // The singleton StoryManager
+    public static StoryManager main;
+
+    // The StoryBeat and coroutine system
+    public List<StoryBeat> StoryBeats;
+    private Coroutine _beatEventsCoroutine;
+
+    // Editor settings
+    public int OpenStoryBeat;
+
+    /// <summary>
+    /// Run initialization functions on the StoryManager
+    /// </summary>
+    private void Awake()
+    {
+        if (main == null)
+        {
+            main = this;
+        }
+        else
+        {
+            Destroy(this);
+        }
+    }
+
+    /// <summary>
+    /// Trigger a story beat by index
+    /// </summary>
+    /// <param name="index">The index of the story beat</param>
+    public void TriggerStoryBeat(int index)
+    {
+        // Edge case if the index is out of bounds
+        if (index < 0 || index >= StoryBeats.Count)
+        {
+            return;
+        }
+
+        // Start the coroutine on the specific story beat
+        _beatEventsCoroutine = StartCoroutine(nameof(TriggerBeatEvents), StoryBeats[index].StoryBeatEvents);
+    }
+
+    /// <summary>
+    /// Start a story beat by name
+    /// </summary>
+    /// <param name="beatName">The name of the story beat</param>
+    public void TriggerStoryBeat(string beatName)
+    {
+        // Iterate through each story beat, activating the one that matches the name
+        foreach (StoryBeat curBeat in StoryBeats)
+        {
+            // Disregard case of the names
+            if (curBeat.Name.ToLower() == beatName.ToLower())
+            {
+                // Start the coroutine on the specific story beat
+                _beatEventsCoroutine = StartCoroutine(nameof(TriggerBeatEvents), curBeat.StoryBeatEvents);
+                return;
+            }
+        }
+    }
+
+    /// <summary>
+    /// Start a specific story beat's events over time
+    /// </summary>
+    /// <param name="events">The events to run</param>
+    /// <returns></returns>
+    private IEnumerator TriggerBeatEvents(List<StoryBeatEvent> events)
+    {
+        WaitForSeconds postEventDelay;
+
+        // Iterate through each event
+        for (int i = 0; i < events.Count; i++)
+        {
+            // Trigger the event, then start a delay before triggering the next event
+            events[i].TriggerEvent();
+            postEventDelay = new(events[i].DelayTime);
```

Comment:
Dunno if making a WaitForSecond variable to store is technically more efficient here since you are making a new one each time. I'm not entirely certain. Nothing needs to change here just commenting